mr/src/main/java/org/elasticsearch/hadoop/util/StringUtils.java

```diff
@@ -409,57 +415,25 @@ public static IpAndPort parseIpAddress(String httpAddr) {
         return new IpAndPort(httpAddr);
     }
 
-    public static String normalize(String path) {
-        if (path == null) {
-            return null;
-        }
-        String pathToUse = path.replace("\\", SLASH);
-
-        int prefixIndex = pathToUse.indexOf(":");
-        String prefix = "";
-        if (prefixIndex != -1) {
-            prefix = pathToUse.substring(0, prefixIndex + 1);
-            if (prefix.contains(SLASH)) {
-                prefix = "";
-            }
-            else {
-                pathToUse = pathToUse.substring(prefixIndex + 1);
-            }
-        }
-        if (pathToUse.startsWith(SLASH)) {
-            prefix = prefix + SLASH;
-            pathToUse = pathToUse.substring(1);
-        }
-
-        List<String> pathList = tokenize(pathToUse, SLASH);
-        List<String> pathTokens = new LinkedList<String>();
-        int tops = 0;
+    public static List<String> normalize(List<URL> urls) throws IOException {
 
-        for (int i = pathList.size() - 1; i >= 0; i--) {
-            String element = pathList.get(i);
-            if (PATH_CURRENT.equals(element)) {
-                // current folder, ignore it
-            }
-            else if (PATH_TOP.equals(element)) {
-                // top folder, skip previous element
-                tops++;
-            }
-            else {
-                if (tops > 0) {
-                    // should it be skipped?
-                    tops--;
-                }
-                else {
-                    pathTokens.add(0, element);
-                }
-            }
+        HashMap<String, String> inoToFile = new HashMap<String, String>();
+        List<String> normalized = new ArrayList<String>();
+        for (URL url: urls) {
+            Path path = Paths.get(url.toString());
+            BasicFileAttributes attr = null;
+            attr = Files.readAttributes(path, BasicFileAttributes.class);
```

Comment:
My understanding is that these API's are java 7 APIs. Right now we're still supporting java 6 in the project. This isn't a hard stop, I'd like to deprecate and remove 6 and 7 support now that the older hadoop versions are removed, but that would have to happen on master before these API's can be used.